runbotics/runbotics-scheduler/src/utils/decorators/user.decorator.ts

```diff
@@ -1,8 +1,12 @@
 import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+import { AuthRequest } from '#/types';
+import { UserEntity } from '#/database/user/user.entity';
 
 export const User = createParamDecorator(
-    (_data: unknown, ctx: ExecutionContext) => {
-        const request = ctx.switchToHttp().getRequest();
-        return request.user;
+    (data: keyof UserEntity | undefined, ctx: ExecutionContext) => {
+        const request = ctx.switchToHttp().getRequest<AuthRequest>();
+        const user = request.user;
+
+        return data ? user?.[data] : user;
```

Comment:
what's that for?

Reply:
I suppose you ask about data param? If it is defined, then is used to pick specific property of user in decorator usage, otherwise whole user is returned.